### PR TITLE
Simplify proguard handling logic.

### PR DESCRIPTION
**Pull request by vkryachko**
This change removes custom logic that configures proguarding and
consumerProguardFiles for SDKs and instead makes all configuration
explicit per SDK.

### CODE REVIEW

File: buildSrc/src/main/java/com/google/firebase/gradle/plugins/FirebaseLibraryPlugin.java [hunk 1]
```diff
@@ -25,6 +25,7 @@
 import com.google.firebase.gradle.plugins.apiinfo.GetMetalavaJarTask;
 import com.google.firebase.gradle.plugins.ci.Coverage;
 import com.google.firebase.gradle.plugins.ci.device.FirebaseTestServer;
+import com.google.firebase.gradle.plugins.license.LicenseResolverPlugin;
```
Comment (rlazo) [inline buildSrc/src/main/java/com/google/firebase/gradle/plugins/FirebaseLibraryPlugin.java:28]: Is this change related to the proguard logic?

File: buildSrc/src/main/java/com/google/firebase/gradle/plugins/license/LicenseResolverPlugin.java [hunk 2]
```diff
@@ -40,7 +40,7 @@
  *       GenerateLicensesTask} to create the artifacts necessary for our release.
  * </ul>
  */
-class LicenseResolverPlugin implements Plugin<Project> {
+public class LicenseResolverPlugin implements Plugin<Project> {
```
Comment (rlazo) [inline buildSrc/src/main/java/com/google/firebase/gradle/plugins/license/LicenseResolverPlugin.java:43]: same

File: buildSrc/src/main/java/com/google/firebase/gradle/plugins/FirebaseLibraryPlugin.java (hunk 1 above)
Comment (vkryachko) [reply buildSrc/src/main/java/com/google/firebase/gradle/plugins/FirebaseLibraryPlugin.java:28]: yes and no :)

It used to live in the root project configuration alongside the proguard logic. So it makes sense to move it here

**Review by rlazo — APPROVED** [review]
